File: lazydev01/day_14.cpp
/*

https://www.codechef.com/problems/MAXCOUNT

*/
#include <bits/stdc++.h>
using namespace std;

#ifdef ONLINE_JUDGE
  #define cerr if(0) cerr
#endif

#define int long long int
#define endl "\n"
#define YES cout<<"YES\n"
#define NO cout<<"NO\n"

int32_t main()
{
ios_base::sync_with_stdio(false);
cin.tie(NULL);

    int t;
    cin >> t;

    while (t--)
    {

        int n;
        cin >> n;
        vector<int> arr(n);
        for(int i=0; i<n; i++){
            cin >> arr[i];
        }
        map<int, int> m;
        for(int i=0; i<n; i++){
            m[arr[i]]++;
        }
        int maximum = 0;
        for(auto i: m){
            if(maximum < i.second){
                maximum = i.second;
            }
        }
        for(auto i: m){
            if(i.second == maximum){
                cout << i.first << " " << i.second << endl;    
                break;
            }
        }
    }
    return 0;
}

File: bansigadhvi1410/day_14.py
"""

https://www.codechef.com/problems/MAXCOUNT


"""


t=int(input())
for _ in range(t):
    n=int(input())
    a=list(map(int,input().split()))
    d={}
    l=[]
    v,c=0,0
    for i in range(len(a)):
        d[a[i]]=0
    for j in range(len(a)):
        d[a[j]]+=1
    for key in d:
        if c<d[key]:
            c=d[key]
    for key in d:
        if d[key]==c:
            l.append(key)
    v=min(l)
    print(v,c)
            